testutil/testproxy/relayerproxy.go

```diff
@@ -219,6 +224,41 @@ func WithDefaultSessionSupplier(
 	}
 }
 
+func WithSuccessiveSessions(
+	supplierKeyName string,
+	serviceId string,
+	appPrivateKey *secp256k1.PrivKey,
+	sessionsCount int,
+) func(*TestBehavior) {
+	return func(test *TestBehavior) {
+		appAddress := GetAddressFromPrivateKey(test, appPrivateKey)
+
+		sessionSuppliers := []string{}
+		var keyring keyringtypes.Keyring
+		err := depinject.Inject(test.Deps, &keyring)
+		require.NoError(test.t, err)
+
+		supplierAccount, err := keyring.Key(supplierKeyName)
+		require.NoError(test.t, err)
+
+		supplierAccAddress, err := supplierAccount.GetAddress()
+		require.NoError(test.t, err)
+
+		supplierAddress := supplierAccAddress.String()
+		sessionSuppliers = append(sessionSuppliers, supplierAddress)
+
+		for i := 0; i < sessionsCount; i++ {
```

Comment:
Please add a comment on what we're achieving here. I don't fully understand how/why this works.